### ISSUE TITLE
Remove deprecated tapconvert and tkunzip in twisted.scripts

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6747]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#6747|
|Type|task|
|Created|2013-09-16 21:34:49Z|
|Branch|https://github.com/twisted/twisted/tree/remove-scripts-6747-2|

`tapconvert` and `tkunzip` in `twisted.scripts` were deprecated in 11.0 ([#5140](https://github.com/twisted/twisted/issues/5140)) and 12.1 ([#5038](https://github.com/twisted/twisted/issues/5038)) and can be removed.

Attachments:

* [6747.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/45a/45a685f772fb83673419208f6c414ff3506ecbc2/bfc5b193f7fa19e612939aac68ef3875910fb647.patch) (14273 bytes) - added by ewong on 2013-11-04 14:26:07Z - 
* [6747.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/45a/45a685f772fb83673419208f6c414ff3506ecbc2/3d394366d4533b3099d9065a6cb04b8b856688aa.patch) (15514 bytes) - added by ewong on 2013-11-05 01:26:48Z - 
* [6747.3.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/45a/45a685f772fb83673419208f6c414ff3506ecbc2/189ec05b5e330bc8a70a0b5fdbe7cd52dc3532bb.patch) (15838 bytes) - added by ewong on 2013-11-05 01:31:42Z - added the topfiles/6747.removal file.
* [6747.4.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/45a/45a685f772fb83673419208f6c414ff3506ecbc2/7cc2963621184f088719f203202db20b7bd72b27.patch) (17377 bytes) - added by ewong on 2013-11-05 05:43:51Z - Removed tapconvert.1 from doc/core/man

<details><summary>Searchable metadata</summary>

```
trac-id__6747 6747
type__task task
reporter__thijs thijs
priority__lowest lowest
milestone__ 
branch__branches_remove_scripts_6747_2 branches/remove-scripts-6747-2
branch_author__hawkowl__ewong hawkowl, ewong
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1379367289000000 1379367289000000
changetime__1429985667409943 1429985667409943
version__None None
owner__hawkowl hawkowl
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="ewong's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ewong set owner to ewong<br>ewong set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="ewong's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ewong set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="ewong's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ewong removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to ewong|
|-|-|

Thanks for your work on this ticket ewong.  Here are a couple comments on the attached patch:

  1. Please read https://twistedmatrix.com/trac/wiki/GitMirror for some tips on using git for Twisted development.  In particular, please set up your git configuration to generate the same kind of patch files svn generates (compatible with -p0 instead of -p1).
  1. The code to deprecate these two modules is (surprisingly) still present.  See `twisted/scripts/__init__.py` for some more code to delete.
  1. Perhaps less surprisingly, `bin/tapconvert` and `bin/tkunzip` are also still present and need to be deleted.

Thanks again for your work on this ticket.

**Comment by twisted-trac** [comment]
|<img alt="ewong's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ewong removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="ewong's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_9"></a>|ewong commented|
|-|-|

6747.3.patch is ready for review.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra set owner to ewong|
|-|-|

Thanks for your patch. There's also still a man page present in [doc/core/man](doc/core/man). Make sure to grep through the code for any references to tapconvert and tkunzip, it's a big codebase.

**Comment by twisted-trac** [comment]
|<img alt="ewong's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ewong removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_13"></a>|@thijstriemstra commented|
|-|-|

(In [40836]) Branching to 'remove-scripts-6747'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_14"></a>|@thijstriemstra commented|
|-|-|

(In [40837]) apply 6747.4.patch, refs [#6747](https://github.com/twisted/twisted/issues/6747)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_15"></a>|@thijstriemstra commented|
|-|-|

(In [40839]) remove zsh definitions, refs [#6747](https://github.com/twisted/twisted/issues/6747)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_16"></a>|@thijstriemstra commented|
|-|-|

Forced a [build](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-scripts-6747). Also came across some zsh definitions that I removed, would be good to have a final review before this gets merged.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_17"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

  1. I'm not sure deleting `twisted/python/zsh/_tapconvert` is the best thing to do.  Note the warning in `twisted/python/zsh/README.txt` about leaving things around for backward-compatibility.  Unfortunately I have no idea for how long we really need to keep these files around.  It would be nice to dig up some more information about this (iow, "we can delete these files when zsh x.y.z is released" or "we can delete these files when distro A, B, and C ship at least zsh x.y.z" or at least "deleting these files will break zsh x.y.z and older" or  something like that).  Until then, perhaps it makes sense not to delete any of these files and instead delete the whole directory when the time comes?  I'm just guessing here, based on `README.txt`.  I'm not a zsh user and have no idea where these files plug in to anything.
  1. `twisted/scripts/tkunzip.py` and `twisted/scripts/tapconvert.py` don't appear to actually be deleted by this branch.  They're just replaced with empty files.  This seems like it's probably a mistake?

Looks fine to me otherwise.  I'd like to say "Please address both points and then merge" but I'm not quite certain enough about the zsh issues.  Perhaps do some zsh digging or get someone who knows about zsh (teratorn?) to add a comment here (or better, add the information to that `README.txt`).

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_18"></a>|@hawkowl commented|
|-|-|

(In [44561]) Branching to remove-scripts-6747-2.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_19"></a>|@hawkowl removed owner|
|-|-|

Merged this forward.

1. Agreed, I've reverted deleting that. It looks like that was all due to be deleted about four years ago, but that's outside of this ticket. They're all unused and should be deleted, looking at http://twistedmatrix.com/trac/ticket/3078 .
2. Made these actually properly deleted.

Builders spun, please review.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_20"></a>|@itamarst set owner to @hawkowl|
|-|-|

Thanks! Merge if Buildbot is happy.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_21"></a>|@hawkowl set status to `closed`|
|-|-|

(In [44565]) Merge remove-scripts-6747-2: Remove tkunzip and tapconvert

Author: ewong, thijs
Reviewers: exarkun, thijs, itamar
Fixes: [#6747](https://github.com/twisted/twisted/issues/6747)
